var/spack/repos/builtin/packages/mpich/package.py

```diff
@@ -551,16 +557,32 @@ def configure_args(self):
         else:
             config_args.append("--with-slurm=no")
 
-        if "pmi=off" in spec:
-            config_args.append("--with-pmi=no")
-        elif "pmi=pmi" in spec:
-            config_args.append("--with-pmi=simple")
-        elif "pmi=pmi2" in spec:
-            config_args.append("--with-pmi=pmi2/simple")
-        elif "pmi=pmix" in spec:
-            config_args.append("--with-pmix={0}".format(spec["pmix"].prefix))
-        elif "pmi=cray" in spec:
-            config_args.append("--with-pmi=cray")
+        # PMI options changed in 4.2.0
+        if spec.satisfies("@4.2:"):
+            # default (no option) is to build both PMIv1 and PMI2 client interfaces
+            if "pmi=pmi1" in spec:
+                # make PMI1 the default client interface
+                config_args.append("--with-pmi=pmi1")
+            elif "pmi=pmi2" in spec:
+                # make PMI2 the default client interface
+                config_args.append("--with-pmi=pmi2")
+            elif "pmi=pmix" in spec:
+                # use the PMIx client interface with an external PMIx library
+                config_args.append("--with-pmi=pmix --with-pmix={0}".format(spec["pmix"].prefix))
+            elif "pmi=cray" in spec:
+                # use PMI2 interface with Cray PMI library
+                config_args.append(
+                    "--with-pmi=pmi2 --with-pmi2={0}".format(spec["cray-pmi"].prefix)
+                )
```

Comment:
While it may work as you wrote it, just in case, it might be safer this way:
```suggestion
                config_args.append("--with-pmi=pmix")
                config_args.append(f"--with-pmix={spec['pmix'].prefix}")
            elif "pmi=cray" in spec:
                # use PMI2 interface of the Cray PMI library
                config_args.append("--with-pmi=pmi2")
                config_args.append(f"--with-pmi2={spec['cray-pmi'].prefix}")
```
PS: Yes, you should use f-strings in `spack` now.